wallet/src/main/scala/org/bitcoins/wallet/DLCWallet.scala

```diff
@@ -332,71 +331,104 @@ abstract class DLCWallet extends LockedWallet with UnlockedWalletApi {
       offerOpt <- dlcOfferDAO.findByEventId(accept.eventId)
       offer = offerOpt.get // Safe, we throw in registerDLCAccept if it is None
       dlc = dlcOpt.get
-      spendingInfoDbs <- listUtxos(offer.fundingInputs.map(_.outPoint))
+      fundingInputs <- dlcInputsDAO.findByEventId(accept.eventId,
+                                                  isInitiator = true)
+      spendingInfoDbs <- listUtxos(fundingInputs.map(_.outPoint))
       spendingInfos = spendingInfoDbs.flatMap(
         _.toUTXOSpendingInfo(keyManager).toSingles)
       client = BinaryOutcomeDLCClient.fromOfferAndAccept(
-        offer.toDLCOffer,
+        offer.toDLCOffer(fundingInputs.map(_.toOutputReference)),
         accept,
         keyManager.rootExtPrivKey.deriveChildPrivKey(account.hdAccount),
         dlc.keyIndex,
         spendingInfos,
-        offer.network)
+        offer.network
+      )
       cetSigs <- client.createCETSigs
       fundingSigs <- client.createFundingTransactionSigs()
-      updatedDLCDb = dlc.copy(initiatorCetSigsOpt = Some(cetSigs),
-                              fundingSigsOpt = Some(fundingSigs))
+      updatedDLCDb = dlc.copy(winSigOpt = Some(cetSigs.winSig),
+                              loseSigOpt = Some(cetSigs.loseSig),
+                              refundSigOpt = Some(cetSigs.refundSig))
       _ <- dlcDAO.update(updatedDLCDb)
     } yield {
       DLCSign(cetSigs, fundingSigs, accept.eventId)
     }
   }
 
+  def addFundingSigs(sign: DLCSign): Future[Vector[DLCFundingInputDb]] = {
```

Comment:
I'm mildly concerned about async-safety here

Reply:
It is probably fine but a comment saying that this is the only way one should insert sigs to the db would be nice